### PR TITLE
fix(autoscaling): can't configure notificationTypes

### PR DESCRIPTION
**Pull request by NetaNir**
----
AutoScalingGroup [notificationconfigurations](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-as-group-notificationconfigurations) property allows configuring autoscaling to send notifications about fleet scaling events to one or more SNS topics. 

The current AutoScalingGroup API expose a `notificationsTopic` property which only allows configuring a single topic, and does not allows configuring which events will trigger a notification but instead configures all notifications, which can be rather noisy.

This PR deprecates the `notificationsTopic` property and introduce a `notifications` property
 which allows configuring multiple `NotificationConfiguration`, each with is own SNS topic and a custom list of events which will trigger a notification.

closes https://github.com/aws/aws-cdk/issues/8053

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 1]
```diff
@@ -766,6 +839,39 @@ export interface RollingUpdateConfiguration {
   readonly suspendProcesses?: ScalingProcess[];
 }
 
+/**
+ * A group of notificationTypes, you can use one of the predefined groups, such as NotificationTypes.ERRORS
+ * or create a custome group by instantiating a `NotificationTypes` object, e.g: `new NotificationTypes(`NotificationType.INSTANCE_LAUNCH`)`.
+ */
+export class NotificationTypes {
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:846]: consider calling this NotificationType (singular) which contains `FleetScalingEvents`.
I know the Cfn properties are called `notificationTypes` and another called `notificationConfigurations`, but I think we can be simpler

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 2]
```diff
@@ -691,6 +718,52 @@ export enum UpdateType {
...
+   * Which fleet scaling events triggers a notification
+   * @default all
+   */
+  readonly notificationTypes?: NotificationTypes;
+}
+
+/**
+ * Fleet scaling event notification type
+ */
+export enum NotificationType {
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:741]: consider calling this `FleetScalingEvents` or something similar to indicate that these are events

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 3]
```diff
@@ -766,6 +839,39 @@ export interface RollingUpdateConfiguration {
...
+  public static readonly ALL = new NotificationTypes(NotificationType.INSTANCE_LAUNCH,
+    NotificationType.INSTANCE_LAUNCH_ERROR,
+    NotificationType.INSTANCE_TERMINATE,
+    NotificationType.INSTANCE_TERMINATE_ERROR);
+
+  /**
+   * Fleet scaling launch events
+   */
+  public static readonly LAUNCH_EVENTS = new NotificationTypes(NotificationType.INSTANCE_LAUNCH, NotificationType.INSTANCE_LAUNCH_ERROR);
+
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:864]: any reason we wouldn't want a predefined type for `TERMINATE_EVENTS` as well? I can envision value in being able to subscribe fleet terminations to a topic

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 4]
```diff
@@ -513,6 +522,23 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
       throw new Error('maxInstanceLifetime must be between 7 and 365 days (inclusive)');
     }
 
+    if (props.notificationsTopic && props.notifications) {
+      throw new Error('Can not set notificationsTopic and notifications, notificationsTopic is deprected use notifications instead');
```
[the remark this answers is not in the document]
Comment (NetaNir) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:526]: thanks!

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts (hunk 1 above)
Comment (NetaNir) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts]: changed to ScalingEvent and ScalingEvents

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts (hunk 3 above)
Comment (NetaNir) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts]: leaving bread crumbs :)

**Pull request merged**